dependencydiff/dependencydiff.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencydiff
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/ossf/scorecard/v4/checks"
+	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/clients/githubrepo"
+	"github.com/ossf/scorecard/v4/clients/githubrepo/roundtripper"
+	"github.com/ossf/scorecard/v4/pkg"
+)
+
+// GetDependencyDiffResults gets dependency changes between two given code commits BASE and HEAD
+// along with the Scorecard check results of the dependencies, and returns a slice of DependencyCheckResult.
+func GetDependencyDiffResults(ownerName, repoName, baseSHA, headSHA string) ([]pkg.DependencyCheckResult, error) {
+	ctx := context.Background()
+	// Fetch dependency diffs using the GitHub Dependency Review API.
+	deps, err := FetchDependencyDiffData(ctx, ownerName, repoName, baseSHA, headSHA)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching dependency changes: %w", err)
+	}
+
+	ghRepo, err := githubrepo.MakeGithubRepo(path.Join(ownerName, repoName))
+	if err != nil {
+		return nil, fmt.Errorf("error creating the github repo: %w", err)
+	}
+	ghRepoClient := githubrepo.CreateGithubRepoClient(ctx, nil)
+	ossFuzzRepoClient, err := githubrepo.CreateOssFuzzRepoClient(ctx, nil)
```

Comment:
do we need this? Something breaks if we set it to `nil`?
Same question for `DefaultVulnerabilitiesClient` below

Reply:
Gotcha. Add a comment.
@azeemsgoogle , was this intended? Shall we file an issue for it?

Reply:
re: do we really want to run all the checks here?

I think for this API, we might want to give all the scorecard check results for users themselves to decide which one to use in their projects. A better way is undoubtedly the [REST query](https://api.securityscorecards.dev/projects/github.com/ossf-tests/scorecard-action), but since we cannot do that yet, running all scorecard checks for every dependency might be an "expediency"?

Reply:
Sure, I'm not against running all the checks. Just something to consider to help scale this better at the cost of removing checks that may not be too impactful.

Reply:
Is this not something the caller will provide? In CLI, via `--checks`, in the Action, via whatever we decide to do there...?

Reply:
Should I add a `chekcsToRun checker.CheckNameToFnMap` param to the API `GetDependencyDiffResults`? 

such that if users specify `--checks` when calling this API, we can run Scorecards on those checks; otherwise we run all the checks?